engine/src/flutter/shell/platform/android/context/android_context.cc

```diff
@@ -10,9 +10,11 @@ AndroidContext::AndroidContext(AndroidRenderingAPI rendering_api)
     : rendering_api_(rendering_api) {}
 
 AndroidContext::~AndroidContext() {
+#if !SLIMPELLER
```

Comment:
Instead of relying on preprocessor macros can you instead have 2 headers with something like this:

```c++
namespace impeller {
  constexpr bool kSlimpellerEnabled = false;
}
```

Then all these preprocessor macros will be scoped instead of hard to maintain preprocessor macros.

This would become
```c++
if (kSlimpellerEnabled) {
  if (main_context_) {
    main_context_->releaseResourcesAndAbandonContext();
  }
}
```

The compiler will just strip them out if it's set to false.

Reply:
We use preprocessor macros / slimpeller for iOS and I'd like to keep it as is. This is not at the point where I think its worth investing in beyond quick fixes

Reply:
I can of course, use gn file exclusdes and change the preprocessor macros a bit to make it nicer to read though.

Reply:
oh, the reason we can't do this is that we don't just rely on dead code elimination: unless we compile with use_ganesh=false, skia itself will retain links to the Skia gpu backend. but setting use_ganesh = false also removes many of these files from the include set at all.

So compilation will fail before we get to that point because headers/types will be missing.